### ISSUE TITLE
[has pr]Non-extendable ProductSearchPageNormalizer with hidden exceptions

### ISSUE THREAD

**Issue by Stopy** [SAP/spartacus#8231]
## Steps to Reproduce
Backend must not send `facets` property in response at `/products/search`

## Observed Results
`OccProductSearchPageNormalizer` is throwing a silent exception while converting in a private method `normalizeUselessFacets`, so `ProductSearchService.getResult()` doesn't emmit a value

## Expected Results
Provide a data anyway

## Repository Used
https://github.com/SAP/spartacus/blob/5b37c2ffb35298a91f6780d1990b4368f8ce321b/projects/core/src/occ/adapters/product/converters/occ-product-search-page-normalizer.ts#L57

## Offer to Fix
Filter facets only if presented.
Also would be good to have a possibility to extend `OccProductSearchPageNormalizer`.

## Additional Information
It would be a good decision to replace built-in normalizer by your own, but built-in normalizer is called anyway, even if you provide your own. Don't know what to do with it.

**Comment by Platonn** [comment]
@Stopy would you mind creating PR with a fix a for filtering? If yes, please take a look on https://sap.github.io/spartacus-docs/commit-guidelines/

Regarding extensibility, did you try providing (i.e. in your app module):
```typescript
{ 
   provide: OccProductSearchPageNormalizer,
   useClass: CustomOccProductSearchPageNormalizer 
}
```
?

**Comment by Stopy** [comment]
@Platonn yes I've tried to provide my custom normalizer, but, as I said, "built-in normalizer is called anyway". But without the error it will satisfy our current requirements. So, yes, I'll create PR. Thanks for your answering

**Comment by Xymmer** [comment]
i marked this issue "backend" but it doesn't seem to have a dependency on backend?
so removed the label

**Comment by LTiger14** [comment]
Fix for #8956

**Comment by LTiger14** [comment]
PR: https://github.com/SAP/spartacus/pull/9407
Testing:
1. Go to `OccProductSearchPageNormalizer`
2. Line 44 replace `target` with `{} as ProductSearchPage`
3. Go to spa and open a PLP, the products should render correctly and the facets should work
